### PR TITLE
Remove common secret config keys and BASE64_NETWORK_CONFIG env

### PR DESCRIPTION
**Pull request by lukaszcl**
This PR removes certain test secrets, such as the Chainlink image and Loki credentials. It also removes support for BASE64_NETWORK_CONFIG as all network secrets are passed via test secrets env vars.

Secrets will now be managed through the test secrets (see https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/config/README.md#test-secrets). 

Related PR in core: https://github.com/smartcontractkit/chainlink/pull/14303

- [x] This PR can be merged after https://github.com/smartcontractkit/ccip/pull/1398 is merged 

---
**Below is a summarization created by an LLM (gpt-4-0125-preview). Be mindful of hallucinations and verify accuracy.**

## Why
The changes streamline the configuration structure across various components by omitting certain fields from TOML serialization and standardizing environment variable handling. This enhances the codebase's clarity and maintainability, especially for external configurations and secret management.

## What
- **config/chainlink_image.go**
  - Changed serialization tag for `Image` field to ignore it during TOML encoding/decoding.
- **config/logging.go**
  - Updated `Loki` and `Grafana` fields to be ignored during TOML encoding/decoding.
  - Modified `LokiConfig` fields (`TenantId`, `Endpoint`, `BasicAuth`, `BearerToken`) to be ignored in TOML serialization.
  - Changed `BearerToken` in `GrafanaConfig` to be ignored in TOML serialization.
- **config/network.go**
  - Added a warning log message about the deprecation of `Base64NetworkConfigEnvVarName`.
- **config/pyroscope.go**
  - Updated fields (`ServerUrl`, `Key`, `Environment`) in `PyroscopeConfig` to be ignored in TOML serialization.
- **config/testconfig_test.go**
  - Added handling for new environment variables related to Pyroscope configuration in test setup and cleanup.
  - Updated expected `PyroscopeConfig` to include `ServerUrl` and `Key` based on environment variables.
- **networks/network_config_helpers_test.go**
  - Removed usage of external TOML parsing for `PyroscopeConfig`, directly initializing the struct instead.
  - Removed the `readPyroscopeConfig` function as it's no longer used.

### CODE REVIEW

**Review by skudasov — APPROVED** [review]

**Review by Tofel — APPROVED** [review]
@lukaszcl approved, but I'd still add some tests to make sure that values we consider secrets are never read via TOML, regardless whether they are passed via base64 override env var or from one of the TOML files.

**Comment by lukaszcl** [comment]
@Tofel 

> but I'd still add some tests to make sure that values we consider secrets are never read via TOML

But this means testing go toml annotations like `toml:"url,omitempty"`? I don't think such tests make sense as this is how the lib/language works. Or maybe you have other tests in mind?

**Comment by lukaszcl** [comment]
@Tofel I updated the known_networks_test.go

File: networks/known_networks_test.go [hunk 1]
```diff
@@ -24,188 +24,6 @@ func TestMustGetSelectedNetworkConfig_MissingSelectedNetwork(t *testing.T) {
...
-	networkCfg := config.NetworkConfig{}
-	err := config.BytesToAnyTomlStruct(l, "test", "", &networkCfg, []byte(testTOML))
-	require.NoError(t, err, "error reading network config")
-
-	require.PanicsWithError(t, fmt.Sprintf("no rpc ws urls found in config for '%s' network", networkName), func() {
-		MustGetSelectedNetworkConfig(&networkCfg)
-	})
-}
-
-func TestMustGetSelectedNetworkConfig_Missing_WalletKeys(t *testing.T) {
```
Comment (lukaszcl) [inline networks/known_networks_test.go:72]: moved this test and related to `TestVariousNetworkConfig`

**Pull request merged**